src/API/Site/Controllers/RestAPI/SyncController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SyncController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI
+ *
+ * @since x.x.x
+ */
+class SyncController extends BaseOptionsController {
+
+	private const DEFAULT_SYNC_MODE = [
+		'products' => [
+			'pull' => false,
+			'push' => false,
+		],
+		'coupons'  => [
+			'pull' => false,
+			'push' => false,
+		],
+		'shipping' => [
+			'pull' => false,
+			'push' => false,
+		],
+		'settings' => [
+			'pull' => false,
+			'push' => false,
+		],
+	];
+
+	/**
+	 * The base for routes in this controller.
+	 *
+	 * @var string
+	 */
+	protected $route_base = 'sync';
+
+	/**
+	 * SyncController constructor.
+	 *
+	 * @param RESTServer $server
+	 */
+	public function __construct( RESTServer $server ) {
+		parent::__construct( $server );
+	}
+
+	/**
+	 * Registers the routes.
+	 */
+	public function register_routes() {
+		$this->register_route(
+			$this->route_base,
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_sync_callback(),
+					'permission_callback' => $this->get_permission_callback(),
```

Comment:
Considering that we only want these endpoints accessible by the API Pull implementation. In the WC endpoints we check for the `gla_syncable` parameter being set to 1 in the request before we modify the responses.

Should we have a custom permissions callback to also check if this is set? Not very difficult to get around, but it might provide better separation.

Reply:
Thanks I will update that in future PRs if that's ok